components/zcash/lib.rs

```diff
@@ -0,0 +1,107 @@
+// Copyright (c) 2024 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+use std::fmt;
+
+use orchard::{
+    zip32::ChildIndex as OrchardChildIndex,
+    keys::Scope as OrchardScope,
+    keys::FullViewingKey as OrchardFVK,
+    zip32::Error as Zip32Error,
+    zip32::ExtendedSpendingKey};
+
+use brave_wallet_cxx::{
```

Comment:
somehow we ended up with this `*_cxx` namespace in other place, but it doesn't follow chromium conventions. We also should be using `brave_wallet` namespace in a `zcash` component, but I also don't think it makes sense to expose this as a public component. This is just an internal implementation in brave wallet, right?

Reply:
Yes it will be used in wallet, but we have filecoin component done same way before.